Swiftz/DictionaryExt.swift

```diff
@@ -6,14 +6,24 @@
 //  Copyright (c) 2014 Maxwell Swadling. All rights reserved.
 //
 
+extension Array {
+	/// Creates a dictionary from the array given a transformer function that returns 
+	/// a (key, value) pair for the dictionary, if nil is returned then the value is
+	/// not added to the dictionary.
+	public func mapKeyValues<K, V>(f : Element -> (K, V)?) -> [K: V] {
```

Comment:
@CodaFi Just did another commit attempting to pull in the idea of label and pairs. I also found a cleaner way to create a Dictionary from a list.